# embedding_dbs

!!! warning

    This section involves juggling conflicting requirements across a large number of protein embedding models. If you want to avoid this, you can just skip this step as embeddings have been [pre-computed for you](/data/#embeddings).

This folder holds all the tools required to build databases of embeddings generated by a number of methods. These databases are used by many of the experiments conducted in the manuscript.

## embed_batch.py
This folder contains the `embed_batch.py` python script, which embeds the sequences of the proteins found in our downstream PPI dataset.

```bash
python embed_batch.py PLLM INPUT_PATH MAX_BATCH_SIZE DEVICE OUTPUT
```

### Arguments

#### PLLM
Specify the pLLM to use to embed protein sequences. The valid options are:

| Command                    | pLLM                               |
|----------------------------|------------------------------------|
| `squeezeprot_sp_nonstrict` | <term:SqueezeProt-SP (Non-strict)> |
| `squeezeprot_sp_strict`    | <term:SqueezeProt-SP (Strict)>     |
| `squeezeprot_u50`          | <term:SqueezeProt-U50>             |
| `prottrans_bert`           | <term:ProtBERT>                    |
| `prottrans_t5`             | <term:ProtT5>                      |
| `proteinbert`              | <term:ProteinBERT>                 |
| `prose`                    | <term:ProSE>                       |
| `esm`                      | <term:ESM>                         |
| `rapppid`                  | <term:RAPPPID>                     |

#### INPUT_PATH
This is the file path to the protein sequences. Valid file formats are:

| Format               | Required Extension |
|----------------------|--------------------|
| [FASTA](https://en.wikipedia.org/wiki/FASTA_format)|`.fasta`|
| [Gzip](https://en.wikipedia.org/wiki/Gzip)'d [FASTA](https://en.wikipedia.org/wiki/FASTA_format)| `.fasta.gz`|
| [RAPPPID PPI Dataset](https://emad-combine-lab.github.io/intrepppid/data.html) |`.h5`|
| [CSV](https://en.wikipedia.org/wiki/Comma-separated_values)| `.csv`|

The CSV format must contain two columns, with the headers "accession" and "sequence", which must contain a protein ID (usually UniProt accession) and the amino acid sequence, respectively. Ultimately, it should look something like:

```
accession,sequence
P05067,MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGR...
Q29537,MEESQSELNIDPPLSQETFSELWNLLPENNVLSSELCPAV...
```

!!! tip

    To encode the sequences from the PPI dataset used in the manuscript, just point this to [`../../data/ppi/rapppid_[common_string_9606.protein.links.detailed.v12.0_upkb.csv]_Mz70T9t-4Y-i6jWD9sEtcjOr0X8=.h5`](/data/).

#### MAX_BATCH_SIZE

The size of the batches of sequences to embed at a time. This will increase RAM/VRAM usage.

#### DEVICE

Either "cpu" or "cuda" depending on whether sequences are to be embedded on the CPU or using a GPU. Defaults to "cpu".

#### OUTPUT

Path to save the embeddings database to. The default paths change with the pLLMs you specify:

| pLLM                               |Path |
|------------------------------------|-----------------------------------------------------|
| <term:SqueezeProt-SP (Non-strict)> |`../../data/embeddings/squeezeprot_sp_nonstrict.lmdb`|
| <term:SqueezeProt-SP (Strict)>     |`../../data/embeddings/squeezeprot_sp_strict.lmdb`   |
| <term:SqueezeProt-U50>             |`../../data/embeddings/squeezeprot_u50.lmdb`         |
| <term:ProtBERT>                    |`../../data/embeddings/prottrans_bert.lmdb`          |
| <term:ProtT5>                      |`../../data/embeddings/prottrans_t5.lmdb`            |
| <term:ProteinBERT>                 |`../../data/embeddings/proteinbert.lmdb`             |
| <term:ProSE>                       |`../../data/embeddings/prose.lmdb`                   |
| <term:ESM>                         |`../../data/embeddings/esm.lmdb`                     |
| <term:RAPPPID>                     |`../../data/embeddings/rapppid.lmdb`                 |

## Requirements

Because some models have conflicting dependencies, we need to create different environments for different models. You will then need to activate/deactivate those environments when running `embed_batch.py` depending on the model you are using.

### RAPPPID

RAPPPID works well with Python 3.8, and the dependencies are in the file `requirements_rapppid.txt`.

More information can be found on the [RAPPPID GitHub repository](https://github.com/jszym/rapppid).

### ProteinBERT

ProteinBERT works well with Python 3.6, and the dependencies are in the file `requirements_proteinbert.txt`. You must then install ProteinBERT proper by cloning the repository and installing thusly:

```bash
git clone https://github.com/nadavbra/protein_bert
cd protein_bert
git submodule init
git submodule update
python setup.py install
```

More information can be found on the [ProteinBERT GitHub repository](https://github.com/nadavbra/protein_bert).

### ProSE

ProSE works well with Python 3.7, and the dependencies are in the file `requirements_prose.txt`.

### All other models

For all other models, we recommend Python 3.11. The dependencies are in the file `requirements.txt`.
